### PR TITLE
feat: add support for SQS trigger

### LINKED ISSUE #18: Support for functions triggered by Amazon SQS
In AWS, one can create a trigger that invokes a Lambda function whenever a queue in Amazon SQS receives input. Technically, the function is invoked with a `Records` array-type property in its `event` argument, containing the messages available in the SQS queue.

In order to support this kind of invocation for a function that can also be called manually with parameters, it would be ideal if the AWS adapter would inspect that argument and fill the provided information into the request body.

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #18

### CODE REVIEW

File: src/aws-adapter.js [hunk 1]
```diff
@@ -152,6 +152,13 @@ async function lambda(evt, ctx) {
       evt.rawQueryString = searchParams.toString();
       evt.headers = {};
       evt.requestContext = { http: {} };
+
+      // function was invoked by a trigger
+      if (!evt.body && evt.Records && evt.Records.length === 1) {
+        evt.requestContext.http.method = 'POST';
+        evt.body = evt.Records[0].body;
+        evt.headers['content-type'] = 'application/json';
+      }
```
Comment (dominique-pfister) [inline src/aws-adapter.js:161]: Deliberately only done for 1 record at a time, so the function will behave as if invoked manually.

**Review by trieloff — APPROVED** [review]

File: src/aws-adapter.js (hunk 1 above)
Comment (tripodsan) [reply src/aws-adapter.js:161]: why not pass in all the records ? there might me actions that can handle several message at once (eg the forms handler)

**Review by tripodsan — CHANGES_REQUESTED** [review]

**Comment by dominique-pfister** [comment]
> why not pass in all the records ? there might me actions that can handle several message at once (eg the forms handler)

The action I intended to use it for includes parameters `owner`, `repo` and `ref`, which are used to load the `fstab` config - if all records are passed or one record in that case as array, the `fstab` loader will no longer work, as it doesn't expect an array, even if it has size of one.

**Comment by tripodsan** [comment]
> > why not pass in all the records ? there might me actions that can handle several message at once (eg the forms handler)
> 
> The action I intended to use it for includes parameters `owner`, `repo` and `ref`, which are used to load the `fstab` config - if records are passed in as array - even if it has length one - the `fstab` loader will no longer work, as it doesn't expect an array.

but this is a very specific use case which should not limit the implementation for all cases.
the fstab loader could be smarter or the action could load the fstab.

alternatively, we could keep your proposal, but also add the records to `context.data` or `context.records` ?

**Comment by dominique-pfister** [comment]
> alternatively, we could keep your proposal, but also add the records to `context.data` or `context.records` ?

I like the `context.records` idea 😄  Instead of adding that specific behaviour for my action to `helix-universal`, it could just add `Records` array to `content.records` and then my action can create a wrapper that:

a) checks `content.records`
b) extracts the first item
c) inserts it into a new request's body (as is done now)
d) invoke the next wrapper in the chain

**Review by tripodsan — APPROVED** [review]

**Pull request merged**